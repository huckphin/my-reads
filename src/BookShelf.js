import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Book from './Book';

class BookShelf extends Component {
    static propTypes = {
        title: PropTypes.string.isRequired,
        books: PropTypes.array.isRequired
    };

    render () {
        const { title, books } = this.props;

        return (
            <div className="bookshelf">
                <h2 className="bookshelf-title">{ title }</h2>
                <div className="bookshelf-books">
                    <ol className="books-grid">
                        {books.length !== 0 && books.map((book => (
                                <li key={ book.title }>
                                    <Book
                                        authors={ book.authors }
                                        imageLinks={ book.imageLinks }
                                        title={ book.title }
                                        key={ book.title } />
                                </li>
                        )))} 
                    </ol>
                </div>
            </div>
        )
    }
}

export default BookShelf;