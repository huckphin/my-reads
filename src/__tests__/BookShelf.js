import React from 'react';
import BookShelf from '../BookShelf';
import renderer from 'react-test-renderer';

test('render a BookShelf', () => {
    const component = renderer.create(
        <BookShelf
            title="Currently Reading"
            key="Currently Reading"
            books={[
                {
                    key: "The Linux Command Line",
                    authors: [ "William E. Shotts, Jr." ],
                    imageLinks: {
                        smallThumbnail: "http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=5&source=gbs_api",
                        thumbnail: "http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api"
                    },
                    title: "The Linux Command Line"
                }
            ]} />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});