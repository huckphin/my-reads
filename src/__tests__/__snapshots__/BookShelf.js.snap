// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render a BookShelf 1`] = `
<div
  className="bookshelf"
>
  <h2
    className="bookshelf-title"
  >
    Currently Reading
  </h2>
  <div
    className="bookshelf-books"
  >
    <ol
      className="books-grid"
    >
      <li>
        <div
          className="book"
        >
          <div
            className="book-top"
          >
            <div
              className="book-cover"
              style={
                Object {
                  "backgroundImage": "url(http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=5&source=gbs_api)",
                }
              }
            />
            <div
              className="book-shelf-changer"
            >
              <select>
                <option
                  disabled={true}
                  value="none"
                >
                  Move to...
                </option>
                <option
                  value="currentlyReading"
                >
                  Currently Reading
                </option>
                <option
                  value="wantToRead"
                >
                  Want to Read
                </option>
                <option
                  value="read"
                >
                  Read
                </option>
                <option
                  value="none"
                >
                  None
                </option>
              </select>
            </div>
          </div>
          <div
            className="book-authors"
          >
            William E. Shotts, Jr.
          </div>
          <div
            className="book-title"
          >
            The Linux Command Line
          </div>
        </div>
      </li>
    </ol>
  </div>
</div>
`;
